subsys/ipc/ipc_service/backends/ipc_rpmsg_static_vrings.c

```diff
@@ -495,3 +495,25 @@ static int backend_init(const struct device *instance)
 			 &backend_ops);
 
 DT_INST_FOREACH_STATUS_OKAY(DEFINE_BACKEND_DEVICE)
+
+#define BACKEND_CONFIG_INIT(n) config[n] = &backend_config_##n;
+
+#if defined(CONFIG_IPC_SERVICE_BACKEND_RPMSG_SHMEM_RESET)
+static int shared_memory_prepare(const struct device *arg)
+{
+	static struct backend_config_t
+		*config[DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT)];
+
+	DT_INST_FOREACH_STATUS_OKAY(BACKEND_CONFIG_INIT);
```

Comment:
How about:
```suggestion
	static const struct backend_config_t *config[] = {
		DT_INST_FOREACH_STATUS_OKAY(BACKEND_CONFIG)
	};
```
with `#define BACKEND_CONFIG(n) &backend_config_##n,`?

Reply:
sure, this change will simplify implementation further

Reply:
fixed